Softmax added to tensor, torch and docs

ShisuiUzumaki opened the pull request:
Fixes #91107 

Added `softmax` docs in 

- `pytorch/torch/_tensor_docs.py`
- `pytorch/torch/_torch_docs.py `
- `pytorch/docs/XXX.rst` files. Here XXX represents all those files where I made the change

Although I have added `softmax` in `docs` directory, I was not sure which files/folders required the edits so there could be issues

cc: @lezcano @albanD 

cc @svekars @carljparker @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen

ShisuiUzumaki committed:
added add_docstr for torch.softmax in pytorch/torch/_torch_docs.py

lezcano commented on docs/source/tensors.rst:
```diff
@@ -665,6 +665,8 @@ Tensor class reference
     Tensor.arctanh
     Tensor.arctanh_
+    Tensor.softmax
+    Tensor.softmax_
```
Does this function exist?

ShisuiUzumaki replied:
> Does this function exist?

are you referring to `Tensor.softmax_`

lezcano commented on torch/_tensor_docs.py:
```diff
@@ -5285,6 +5285,15 @@ def callable(a, b) -> number
+    r"""
+softmax(dim) -> Tensor
+
+See :func:`torch.softmax`
```
point to `torch.nn.funcional.softmax`.

lezcano commented on docs/source/torch.rst:
```diff
@@ -385,6 +385,7 @@ Pointwise Ops
     subtract
     tan
     tanh
+    softmax
```
In alphabetical order please.

lezcano commented on docs/source/tensors.rst:
```diff
@@ -665,6 +665,8 @@ Tensor class reference
     Tensor.atanh_
     Tensor.arctanh
     Tensor.arctanh_
+    Tensor.softmax
```
alphabetical order unless there's a good reason.

lezcano commented on docs/source/name_inference.rst:
```diff
@@ -222,6 +222,7 @@ If you don't see an operation listed here, but it would help your use case, plea
    :meth:`Tensor.tan_`,None
    ":meth:`Tensor.tanh`, :func:`torch.tanh`",:ref:`keeps_input_names-doc`
    :meth:`Tensor.tanh_`,None
+   ":meth:`Tensor.softmax`, :func:`torch.tanh`",:ref:`keeps_input_names-doc`
```
alphabetical order.

lezcano replied:
also, please be careful when copy pasting.
```suggestion
   ":meth:`Tensor.softmax`, :func:`torch.softmax`",:ref:`keeps_input_names-doc`
```

ShisuiUzumaki commented:
@lezcano Please review if I have made the required changes correctly

lezcano commented:
the commit is polluted

lezcano reviewed (approved):
Cool! Now you just have to sign the CLA and we should be good to go.
If you signed it already, I think that your commits may have got polluted, so squashing them / reseting the author may help.

ShisuiUzumaki commented:
> Cool! Now you just have to sign the CLA and we should be good to go. If you signed it already, I think that your commits may have got polluted, so squashing them / reseting the author may help.

I am having some trouble with this. I have signed the CLA so many times and tried changing author using `git rebase`, but it does seem to work

lezcano commented:
Consider squashing the PRs and then doing `git commit --ame --reset-author`

lezcano commented:
/easycla

ShisuiUzumaki committed:
parent a6257045a65c5f7e13eba34ef907676b1f1be486
author ShisuiUzumaki <hsalahhuddin@gmail.com> 1671602557 +0500
committer Salahuddin <60926009+ShisuiUzumaki@users.noreply.github.com> 1671810286 +0000

added add_docstr_all for softmax in pytorch/torch/_tensor_docs.py

softmax added in *.rst files in docs folder

Tensor object has not attribute softmax_ error fixed

issues fixed for softmax/softmax_ in torch and docs directories

issues fixed for softmax/softmax_ in torch and docs directories

ShisuiUzumaki commented:
@lezcano For some reason, now CircleCI build is failing. Do you have any idea?

ShisuiUzumaki committed:
Merge branch 'pytorch:master' into softmax_to_docs

ShisuiUzumaki commented:
> /easycla

I think all the checks are clear now @lezcano

lezcano commented:
@pytorchbot merge -f "just docs"

ShisuiUzumaki commented:
> @pytorchbot merge -f "just docs"

Thanks for all the help @lezcano it was a great learning experience 😊

weiwangmeta commented:
@pytorchbot revert -m "breaking internal distributed testing builds" -c ghfirst

lezcano commented:
wow, that's unexpected. Do you have more details @weiwangmeta?

ShisuiUzumaki commented:
> wow, that's unexpected. Do you have more details @weiwangmeta?

Yes, if there are any changes needed, please let me know

weiwangmeta commented:
@ShisuiUzumaki  @lezcano Looks like the internal distributed test may have some non-deterministic failure. Let's plan on getting this in again.

weiwangmeta commented:
@pytorchbot merge -f "just docs"

lezcano commented:
@pytorchbot merge -f "just docs"
